src/cli.rs

```diff
@@ -122,12 +122,51 @@ fn print_chips() {
 
 /// The string reported by the `--version` flag
 fn print_version() {
-    const VERSION: &str = env!("CARGO_PKG_VERSION"); // version from Cargo.toml e.g. "0.1.4"
-    const HASH: &str = include_str!(concat!(env!("OUT_DIR"), "/git-info.txt")); // "" OR git hash e.g. "34019f8" -- this is generated in build.rs
+    /// Version from `Cargo.toml` e.g. `"0.1.4"`
+    const VERSION: &str = env!("CARGO_PKG_VERSION");
+
+    /// `""` OR git hash e.g. `"34019f8"`
+    ///
+    /// `git describe`-docs:
+    /// > The command finds the most recent tag that is reachable from a commit. (...)
+    /// It suffixes the tag name with the number of additional commits on top of the tagged object
+    /// and the abbreviated object name of the most recent commit.
+    //
+    // The `fallback` is `"--"`, cause this will result in "" after `fn extract_git_hash`.
+    const GIT_DESCRIBE: &str = git_version!(fallback = "--", args = ["--long"]);
+    // Extract the "abbreviated object name"
+    let hash = extract_git_hash(GIT_DESCRIBE);
+
     println!(
-        "{}{}\nsupported defmt version: {}",
-        VERSION,
-        HASH,
-        defmt_decoder::DEFMT_VERSION
+        "{} {}\nsupported defmt version: {}",
+        VERSION, hash, DEFMT_VERSION
     );
 }
+
+/// Extract git hash from a `git describe` statement
+fn extract_git_hash(git_describe: &str) -> &str {
+    git_describe.split("-").nth(2).unwrap()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn should_extract_hash_from_description() {
+        let hash = extract_git_hash("v0.2.3-12-g25c50d2");
```

Comment:
what happens if there are no extra commits yet after a fresh tag, e.g. a `git describe` yields output like `v0.2.3`?

Reply:
It won't only give us the tag, but the long format, since we specify `args = ["--long"]`.

```
       --long
           Always output the long format (the tag, the number of commits and
           the abbreviated commit name) even when it matches a tag. This is
           useful when you want to see parts of the commit object name in
           "describe" output, even when the commit in question happens to be a
           tagged version. Instead of just emitting the tag name, it will
           describe such a commit as v1.2-0-gdeadbee (0th commit since tag
           v1.2 that points at object deadbee....).
```

Reply:
aaah perfect